File: tooling/ingest-python/tests/xadapter_sample.py
"""Cross-adapter fixture (Python side). Each function is written to translate to the SAME Nova
Lingua body as its Rust twin in `xadapter_sample.rs`, so the two adapters must agree byte-for-byte
on the body content-address. Consumed by `TestCrossAdapterAgreement`."""


def double(n: int) -> int:
    """Double a number.

    >>> double(5)
    10
    """
    return n + n


def times2(n: int) -> int:
    """Twice a number (a literal operand).

    >>> times2(5)
    10
    """
    return n * 2


def is_pos(n: int) -> bool:
    """Whether a number is positive.

    >>> is_pos(3)
    True
    >>> is_pos(-1)
    False
    """
    return n > 0


def safe_div(a: int, b: int) -> int | None:
    """Divide, or nothing on a zero divisor (a Maybe: None vs Some/Just).

    >>> safe_div(6, 2)
    3
    >>> safe_div(7, 0)
    """
    if b == 0:
        return None
    return a // b


def add_sub(a: int, b: int) -> tuple[int, int]:
    """The sum and the difference (a tuple result).

    >>> add_sub(5, 3)
    (8, 2)
    """
    return (a + b, a - b)

File: tooling/ingest-python/tests/xadapter_sample.rs
//! Cross-adapter fixture (Rust side). Each function is written to translate to the SAME Nova Lingua
//! body as its Python twin in `xadapter_sample.py`, so the two adapters must agree byte-for-byte on
//! the body content-address. Consumed by `TestCrossAdapterAgreement`.

/// Double a number.
/// ```
/// assert_eq!(double(5), 10);
/// ```
pub fn double(n: i64) -> i64 {
    n + n
}

/// Twice a number (a literal operand).
/// ```
/// assert_eq!(times2(5), 10);
/// ```
pub fn times2(n: i64) -> i64 {
    n * 2
}

/// Whether a number is positive.
/// ```
/// assert_eq!(is_pos(3), true);
/// assert_eq!(is_pos(-1), false);
/// ```
pub fn is_pos(n: i64) -> bool {
    n > 0
}

/// Divide, or nothing on a zero divisor (a Maybe: None vs Some/Just).
/// ```
/// assert_eq!(safe_div(6, 2), Some(3));
/// assert_eq!(safe_div(7, 0), None);
/// ```
pub fn safe_div(a: i64, b: i64) -> Option<i64> {
    if b == 0 {
        None
    } else {
        Some(a / b)
    }
}

/// The sum and the difference (a tuple result).
/// ```
/// assert_eq!(add_sub(5, 3), (8, 2));
/// ```
pub fn add_sub(a: i64, b: i64) -> (i64, i64) {
    (a + b, a - b)
}
